### PR TITLE
feat: LOPS-127: Update frontend SemanticSearch component to leverage views and tasks endpoint

### PR DESCRIPTION
**Pull request by yyassi-heartex**
### PR fulfills these requirements
- [x] Commit message(s) and PR title follows the format `[fix|feat|ci|chore|doc]: TICKET-ID: Short description of change made` ex. `fix: DEV-XXXX: Removed inconsistent code usage causing intermittent errors`
- [ ] Tests for the changes have been added/updated (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [ ] Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
- [x] Self-reviewed and ran all changes on a local instance (for bug fixes/features)

#### Change has impacts in these area(s)
_(check all that apply)_
- [ ] Product design
- [x] Frontend

### Describe the reason for change
_(link to issue, supportive screenshots etc.)_

#### What does this fix?
_(if this is a bug fix)_

#### What is the new behavior?
_(if this is a breaking or feature change)_

#### What is the current behavior?
_(if this is a breaking or feature change)_

#### What libraries were added/updated?
_(list all with version changes)_

#### Does this change affect performance?
_(if so describe the impacts positive or negative)_

#### Does this change affect security?
_(if so describe the impacts positive or negative)_

#### What alternative approaches were there?
_(briefly list any if applicable)_

#### What feature flags were used to cover this change?
_(briefly list any if applicable)_

### Does this PR introduce a breaking change?
_(check only one)_
- [ ] Yes, and covered entirely by feature flag(s)
- [ ] Yes, and covered partially by feature flag(s)
- [ ] No
- [ ] Not sure (briefly explain the situation below)

### What level of testing was included in the change?
_(check all that apply)_
- [ ] e2e
- [ ] integration
- [ ] unit

### Which logical domain(s) does this change affect?
_(for bug fixes/features, be as precise as possible. ex. Authentication, Annotation History, Review Stream etc.)_

### CODE REVIEW

File: src/stores/Tabs/tab.js [hunk 1]
```diff
@@ -16,6 +16,12 @@ import { TabSelectedItems } from "./tab_selected_items";
 import { History } from '../../utils/history';
 import { FF_LOPS_12, isFF } from "../../utils/feature-flags";
 
+const ThresholdModel = types
+  .model("ThresholdModel", {
+    from: types.optional(types.number, 0),
+    to: types.optional(types.number, 100),
+  });
+
```
Comment (bmartel) [inline src/stores/Tabs/tab.js:24]: Is there any reason not to move this out to its own model? I don't see much relation to Tabs here, all the data is separate and should be separate to Datamanager.

**Review by bmartel — CHANGES_REQUESTED** [review]

File: src/stores/Tabs/tab.js (hunk 1 above)
Comment (yyassi-heartex) [reply src/stores/Tabs/tab.js:24]: no reason not to move this into a separate model file - we just dont use it else where - will do

File: src/components/CandidateTaskView/CandidateTaskView.js [hunk 2]
```diff
@@ -5,14 +5,19 @@ import "./CandidateTaskView.styl";
 import { getRoot } from "mobx-state-tree";
 import { useEffect, useState } from "react";
 import { format } from "date-fns";
+import { FF_LSDV_4711, isFF } from "../../utils/feature-flags";
+
+const imgDefaultProps = {};
```
Comment (nicholasrq) [inline src/components/CandidateTaskView/CandidateTaskView.js:10]: it's better to uppercase top-level constants

```suggestion
const IMG_DEFAULT_PROPS = {};
```

**Review by nicholasrq — APPROVED** [review]

File: src/stores/Tabs/tab.js [hunk 3]
```diff
@@ -296,6 +298,11 @@ export const Tab = types
       self.selected = ids;
     },
 
+    setSemanticSearch(searchText) {
```
Comment (bmartel) [inline src/stores/Tabs/tab.js:301]:
```suggestion
    setSearchText(searchText) {
```

Comment (bmartel) [reply src/stores/Tabs/tab.js:301]: I would keep this open here, I think this search text usage is something best defined by the extension.

**Review by bmartel — CHANGES_REQUESTED** [review]

**Review by bmartel — APPROVED** [review]

**Pull request merged**